capabilities/istio/pepr/istio-job-termination.ts

```diff
@@ -9,39 +9,60 @@ export const IstioJobTermination = new Capability({
 // Use the 'When' function to create a new action
 const { When } = IstioJobTermination;
 
+// Keep track of in-progress terminations
+const inProgress: Record<string, boolean> = {};
+
 When(a.Pod)
-  .IsCreatedOrUpdated() // IsCreated doesn't trigger enough :thinking:
+  .IsUpdated()
   .WithLabel("batch.kubernetes.io/job-name")
   .WithLabel("service.istio.io/canonical-name")
   .Watch(async pod => {
-    if (pod.status.phase == "Running") {
-      const podReadyForTermination = R.all(containerStatus => {
-        return (
-          containerStatus.state.terminated?.exitCode == 0 ||
-          containerStatus.name == "istio-proxy"
-        );
-      })(pod.status.containerStatuses);
-
-      if (podReadyForTermination) {
-        Log.info("Attempting to terminate sidecar for " + pod.metadata.name);
+    const { metadata, status } = pod;
+    const { name, namespace } = metadata;
+
+    // Ensure termination isn't already in progress
+    if (inProgress[name]) {
+      return;
+    }
+
+    // Only terminate if the pod is running
+    if (status.phase == "Running") {
+      // Check if the pod has a non-istio container that has terminated
+      const canTerminate = !!status?.containerStatuses?.find(
+        ({ name, state }) => name != "istio-proxy" && state?.terminated,
+      );
+
+      if (canTerminate) {
+        // Mark the pod as seen
+        inProgress[name] = true;
```

Comment:
`inProgress` probably needs to be keyed off of `namespace/name` to ensure it is unique across the cluster